### PR TITLE
feat: reconnect to service logs and link

### PR DESCRIPTION
**Pull request by adschwartz**
## Description:
* automatically reconnect service logs when connection fails
* add link to get to cloud ui when running in cloud context

## Is this change user facing?
YES

### CODE REVIEW

File: enclave-manager/web/src/client/enclaveManager/KurtosisClient.ts [hunk 1]
```diff
@@ -54,6 +55,10 @@ export abstract class KurtosisClient {
     console.log("baseApplicationUrl", this.baseApplicationUrl);
   }
 
+  isRunningInCloud() {
+    return this.cloudUrl.host.toLowerCase().includes(KURTOSIS_CLOUD_HOST);
+  }
```
Comment (Dartoxian) [inline enclave-manager/web/src/client/enclaveManager/KurtosisClient.ts:60]: I think I'd have probably just made this an abstract method, and then in the local client always return false, and in the authenticated client return true

File: enclave-manager/web/src/emui/Navbar.tsx [hunk 2]
```diff
@@ -1,12 +1,15 @@
...
+import { KURTOSIS_CLOUD_CONNECT_URL } from "../client/constants";
 import { NavButton, Navigation } from "../components/Navigation";
 
 export type NavbarProps = {
   baseApplicationUrl: URL;
+  isRunningInCloud: boolean;
 };
 
-export const Navbar = ({ baseApplicationUrl }: NavbarProps) => {
+export const Navbar = ({ isRunningInCloud, baseApplicationUrl }: NavbarProps) => {
```
Comment (Dartoxian) [inline enclave-manager/web/src/emui/Navbar.tsx]: no need to thread `isRunningInCloud` through - just `useKurtosisClient` here.

File: enclave-manager/web/src/emui/enclaves/enclave/service/logs/ServiceLogs.tsx [hunk 3]
```diff
@@ -20,37 +20,55 @@ type ServiceLogsProps = {
   service: ServiceInfo;
 };
 
+// @ts-ignore
+export async function reTryCatch(callback, times: number = 1, isRetry: boolean = false) {
```
Comment (Dartoxian) [inline enclave-manager/web/src/emui/enclaves/enclave/service/logs/ServiceLogs.tsx]:
```suggestion
export async function reTryCatch<R>(
  callback: (isRetry: boolean) => Promise<R>,
  times: number = 1,
  isRetry: boolean = false,
): Promise<R> {
```

File: enclave-manager/web/src/emui/enclaves/enclave/service/logs/ServiceLogs.tsx [hunk 4]
```diff
@@ -20,37 +20,55 @@ type ServiceLogsProps = {
...
+  }
+}
+
 export const ServiceLogs = ({ enclave, service }: ServiceLogsProps) => {
   const kurtosisClient = useKurtosisClient();
   const [logLines, setLogLines] = useState<LogLineProps[]>([]);
 
   useEffect(() => {
-    let cancelled = false;
+    let canceled = false;
```
Comment (Dartoxian) [inline enclave-manager/web/src/emui/enclaves/enclave/service/logs/ServiceLogs.tsx:46]: lol

File: enclave-manager/web/src/emui/enclaves/enclave/service/logs/ServiceLogs.tsx [hunk 5]
```diff
@@ -20,37 +20,55 @@ type ServiceLogsProps = {
...
     const abortController = new AbortController();
-    (async () => {
+    if (kurtosisClient) {
```
Comment (Dartoxian) [inline enclave-manager/web/src/emui/enclaves/enclave/service/logs/ServiceLogs.tsx]: we are guaranteed to have a `kurtosisClient` - `useKurtosisClient` always provides us an instantiated, ready to go, client - so this check isn't needed.

**Review by Dartoxian — APPROVED** [review]

File: enclave-manager/web/src/emui/Navbar.tsx (hunk 2 above)
Comment (adschwartz) [reply enclave-manager/web/src/emui/Navbar.tsx]: Good idea!

File: enclave-manager/web/src/client/enclaveManager/KurtosisClient.ts (hunk 1 above)
Comment (adschwartz) [reply enclave-manager/web/src/client/enclaveManager/KurtosisClient.ts:60]: good idea!

**Pull request merged**